docs/src/extend/custom-rules.md

```diff
@@ -614,17 +614,40 @@ You can also access comments through many of `sourceCode`'s methods using the `i
 
 Rules may export a `schema` property, which is a [JSON Schema](https://json-schema.org/) format description of a rule's options which will be used by ESLint to validate configuration options and prevent invalid or unexpected inputs before they are passed to the rule in `context.options`.
 
-There are two formats for a rule's exported `schema`:
+If `schema` is not specified, then any options (besides severity) supplied to the rule are passed on verbatim without validation.
```

Comment:
Just want to call out that some of this behavior will be changing in the next major release v9 due to [RFC-85](https://github.com/eslint/rfcs/tree/main/designs/2021-schema-object-rules) ([draft PR](https://github.com/eslint/eslint/pull/16614)).


>- Requiring rules with options to specify schemas (removing support for rules with options that are missing schemas)
>- Requiring rule implementations to use the object-style format (removing support for the deprecated function-style format)

It would ideal if we can take that into account here by:
* Avoid providing recommendations that conflict with the upcoming requirements
* Write the documentation in such a way that it can be easily updated when the new requirements come into effect

Reply:
> Avoid providing recommendations that conflict with the upcoming requirements

I don't think the docs as written now _recommend_ not providing a schema, it just explains the consequence of not providing one. I can update the line to note that it is explicitly _not_ recommended, and that this option will be removed in v9. Once v9 drops, the change to the docs should be as simple as replacing that paragraph to explain that if you don't provide a `schema` object, your rule will not be permitted to have any options specified (equivalent to `[]`)?

> Requiring rule implementations to use the object-style format (removing support for the deprecated function-style format)

I don't think these docs have any conflict with that? They explain `schema`/`meta.schema`, but that change is eliminating the `const rule = function(context, options) { ... }; rule.schema = { ... };` form afaict, which is no longer documented at all?

Reply:
Updated docs to:

>If `schema` is not specified, then any options (besides severity) supplied to the rule are passed on verbatim without validation. This is a legacy behaviour though, and all rules are encouraged to specify a schema to aid developers when configuring your rules. Note that in v9, this behaviour will be deprecated entirely, and not specifying `schema` will force the rule to have no options (equivalent to `[]`) - see the [Require schemas and object-style rules](https://github.com/eslint/rfcs/blob/main/designs/2021-schema-object-rules/README.md) RFC.

Does this cover the bases you wanted to cover?

Reply:
Also noting in the PR you linked:

> Throw with no-op schema schema: {}

Would it be helpful to add some further rule schema validations to catch obvious cases that will _always_ fail, like a top-level `type: 'object/number/string'`? Could also recurse into `anyOf`/`oneOf`/`allOf` cases and do the same check - i.e. the first `type` to be tested always needs to be an `array`?

Not necessarily suggesting adding those validations in this PR, though I'm happy to take a stab at it separately?

Reply:
Your documentation tweaks sound reasonable, thanks!

As for catching more bad schemas, I certainly wouldn't mind that, but out of scope for my initial [PR](https://github.com/eslint/eslint/pull/16614) implementing that [RFC](https://github.com/eslint/rfcs/tree/main/designs/2021-schema-object-rules). I think we could consider another breaking change PR with more validation though.